workflows/common/python/model_runner.py

```diff
@@ -171,50 +171,62 @@ def run(hyper_parameter_map, obj_return):
-        info = sys.exc_info()
-        s = traceback.format_tb(info[2])
-        sys.stdout.write('\\n\\nEXCEPTION in model run(): \\n' +
-                         repr(e) + ' ... \\n' + ''.join(s))
-        sys.stdout.write('\\n')
-        sys.stdout.flush()
-
-        # logger.warn("Caught InvalidArgumentError")
-        exception = True
-        exit(1)
-    log("PKG RUN STOP")
-
-    if framework == "keras":
-        runner_utils.keras_clear_session(framework)
-
-    stop_perf(Ps)
-    finish = time.time()
-    duration = finish - start
 
     # check for epochs if not present set to 1, used for checking early stopping in function get_results
     if "epochs" in hyper_parameter_map:
         epochs = hyper_parameter_map["epochs"]
     else:
         epochs = 1
+        
+    if framework == 'keras':
+
+        try:
+            history = pkg.run(params)
+        except Exception as e:
+            logger.warn("RUN EXCEPTION: " + str(e))
+            print("RUN EXCEPTION: " + str(e))
+            info = sys.exc_info()
+            s = traceback.format_tb(info[2])
+            sys.stdout.write('\\n\\nEXCEPTION in model run(): \\n' +
+                            repr(e) + ' ... \\n' + ''.join(s))
+            sys.stdout.write('\\n')
+            sys.stdout.flush()
+
+            # logger.warn("Caught InvalidArgumentError")
+            exception = True
+            exit(1)
+        log("PKG RUN STOP")
+
+        # if framework == "keras":
+        runner_utils.keras_clear_session(framework)
 
-    # Default result if there is no val_loss (as in infer.py)
-    result = 0
-    history_result = {}
-    if not exception:
-        logger.info("DONE: run_id %s in %0.2f seconds." %
-                    (hyper_parameter_map["run_id"], duration))
-        if history is not None:
-            if history == "EPOCHS_COMPLETED_ALREADY":
-                result, history_result = "EPOCHS_COMPLETED_ALREADY", None
-            else:
-                result, history_result = get_results(history, obj_return,
-                                                     epochs)
-    else:
-        result, history_result = "RUN_EXCEPTION", None
+        stop_perf(Ps)
+        finish = time.time()
+        duration = finish - start
+
+        # Default result if there is no val_loss (as in infer.py)
+        result = 0
+        history_result = {}
+        if not exception:
+            logger.info("DONE: run_id %s in %0.2f seconds." %
+                        (hyper_parameter_map["run_id"], duration))
+            if history is not None:
+                if history == "EPOCHS_COMPLETED_ALREADY":
+                    result, history_result = "EPOCHS_COMPLETED_ALREADY", None
+                else:
+                    result, history_result = get_results(history, obj_return,
+                                                        epochs)
+        else:
+            result, history_result = "RUN_EXCEPTION", None
+
+    elif framework == 'pytorch':
```

Comment:
Good changes here, will this be true for any pytorch model?

It'd be good to launch our Jenkins testing on this branch before merging.

Reply:
This is not pytorch specific. We could also have this for keras. This change assumes that after training the model, the run() method returns prediction scores like loss, rmse for validation and test sets. We can have this for all the models, but currently not every drug response model returns these values.